Fixed dockerfile for docker containerization

RyanMarinoff opened the pull request:
Fixed the docker file so to allow containerization of the backend.

RyanMarinoff committed:
Fixed dockerfile for docker containerization

AdamRev reviewed (commented):
If you want to add an EXPOSE that's fine, but why hardcode the name of the jarfile (with version and snapshot) instead of using the argument?

rlayneorr reviewed (changes requested):
Change back to using ARG

RyanMarinoff committed:
new fix for Dockerfile: successful commands: Sending build context to Docker daemon   44.5MB
Step 1/6 : FROM openjdk:8-jdk-alpine
 ---> a3562aa0b991
Step 2/6 : VOLUME /tmp
 ---> Using cache
 ---> 71e415f2921e
Step 3/6 : ARG JAR_FILE
 ---> Using cache
 ---> da8780805740
Step 4/6 : COPY $JAR_FILE ask-backend.jar
 ---> Using cache
 ---> ec082f5db2cb
Step 5/6 : CMD ["java","-jar", "ask-backend.jar"]
 ---> Using cache
 ---> 29251d9c575e
Step 6/6 : EXPOSE 1337/tcp
 ---> Using cache
 ---> b1faaa9521ca
Successfully built b1faaa9521ca
SECURITY WARNING: You are building a Docker image from Windows against a non-Windows Docker host. All files and directories added to build context will have '-rwxr-xr-x' permissions. It is recommended to double check and reset permissions for sensitive files and directories.docker container run -p 127.0.0.1:1337:1337/tcp b1faaa9521ca {replace b1faaa9521ca with build name in 'Successfully built b1faaa9521ca'}

RyanMarinoff commented:
Changes are made to allow for use of ARG. Success build and deploy on local system with the following commands:

`docker build --build-arg JAR_FILE=./target/ask-0.0.1-SNAPSHOT.jar .`

Output:

```
...snip...
Step 6/6 : EXPOSE 1337/tcp
 ---> Using cache
 ---> b1faaa9521ca
Successfully built b1faaa9521ca
...snip...
```
`docker container run -p 127.0.0.1:1337:1337/tcp b1faaa9521ca`

Note: replace `b1faaa9521ca` with your successful build ID.

The previous hardcoding the jar path was due to not implementing ARG correctly while creating this fix.
